### PR TITLE
refactor: 💡 refactor managed grouped mirage data for LDAP

### PR DESCRIPTION
**Pull request by dvonanderson**
✅ Closes: ICU-10055

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-10055)

## Description
Added LDAP attributes for managed groups

### Screenshots (if appropriate):

Screenshot 2023-07-14 at 09 42 01

### CODE REVIEW

File: addons/api/mirage/factories/managed-group.js [hunk 1]
```diff
@@ -23,12 +24,18 @@ export default factory.extend({
...
   attributes() {
     switch (this.type) {
-      case 'oidc':
+      case TYPE_AUTH_METHOD_OIDC:
         return {
           filter: faker.random.words(),
         };
+      case TYPE_AUTH_METHOD_LDAP:
+        return {
+          group_names: [faker.internet.userName({ min: 1, max: 5 })],
```
Comment (lisbet-alvarez) [inline addons/api/mirage/factories/managed-group.js:37]: i think `faker.random.word` might work better here.  Also I don't think [userName](https://v7.fakerjs.dev/api/internet.html#username) has the min and max option

Comment (dvonanderson) [reply addons/api/mirage/factories/managed-group.js]: @lisbet-alvarez good call, fixed

File: addons/api/mirage/factories/managed-group.js [hunk 2]
```diff
@@ -23,12 +24,24 @@ export default factory.extend({
...
   attributes() {
     switch (this.type) {
-      case 'oidc':
+      case TYPE_AUTH_METHOD_OIDC:
         return {
           filter: faker.random.words(),
         };
+      case TYPE_AUTH_METHOD_LDAP: {
+        const num = faker.datatype.number({ min: 2, max: 4 });
+        let names = [];
```
Comment (ZedLi) [inline addons/api/mirage/factories/managed-group.js:37]:
```suggestion
        const names = [];
```

Comment (dvonanderson) [reply addons/api/mirage/factories/managed-group.js:37]: @ZedLi thank you for that!

**Review by DhariniJeeva — APPROVED** [review]

**Review by lisbet-alvarez — APPROVED** [review]

**Pull request merged**